test/quantum_objects.jl

```diff
@@ -383,4 +386,22 @@
     @test_throws ArgumentError permute(bra_bdca, wrong_order2)
     @test_throws ArgumentError permute(op_bdca, wrong_order1)
     @test_throws ArgumentError permute(op_bdca, wrong_order2)
+
+    @testset "Type Inference" begin
+        for T in [Float32, Float64, ComplexF32, ComplexF64]
+            N = 25
+            a = rand(T, N)
```

Comment:
Maybe also add a direct test (without specifying `type`, like the one mentioned in issue #177 ) before the for-loop ?

something like
```julia
for T in [Float32, Float64, ComplexF32, ComplexF64]
    N = 25
    a = rand(T, N)
    @inferred Qobj(a)
    for type in [Ket, OperatorKet]
       ...
```